File: tests/fixtures/generic/vulnerable/taint_xpathi.py
# @audit-fixture
# @rule: taint_xpathi
# @category: security
# @expected: detected
from flask import request
from lxml import etree
@app.route("/search")
def search():
    query = request.args.get("q")
    tree.xpath(f"//item[name='{query}']")

File: tests/fixtures/generic/vulnerable/taint_xpathi.php
<?php
// @audit-fixture
// @rule: taint_xpathi
// @expected: detected
$dom = new DOMDocument();
$dom->loadXML($xmlData);
$xpath = new DOMXPath($dom);
$results = $xpath->query("//user[@name='" . $_GET['name'] . "']");
